% !Rnw root = Cockell_main.Rnw
\section{Introduction} % section title
\label{SC1-sect:intro}  % \label{} allows reference to this section
%!!! please use unique labels (e.g., include your initials) for all
%your sections, equations, figures, tables, etc.

\subsection{What is Bioconductor?}

Bioconductor \cite{Gentleman04} is an open source software project for the R statistical computing language.  The main aim of the project is to provide R packages to facilitate the analysis of DNA microarray, sequencing, SNP and other genomic data.  In addition to various packages for analysis of data, they also distribute meta-data packages that provide useful annotation.  The stable non-development version of Bioconductor is normally released twice annually\footnote{http://www.bioconductor.org/news/bioc\_2\_11\_release/}. In addition to providing packages the Bioconductor site (\href{http://www.bioconductor.org}{bioconductor.org}) also provides documentation for each package, often a brief vignette and a more comprehensive user guide.

\subsection{Illumina microarrays}

The Illumina BeadArray\textsuperscript{TM} microarray platform makes use of BeadChips, which, as their name suggests are composed of tiny beads.  These beads are made of silica and are just 3$\mu$m in diameter, and are covered in hundreds of thousands of 79 nucleotide long oligonucleotide sequences; individual beads each having a different probe sequence.  A unique property of BeadChips, which distinguish them from Affymetrix arrays, is that every array is unique, the beads self assemble into microwells found on the slide.  As a consequence of this, most probes will be present on the array \~30 times \cite{Gunderson04}. An additional consequence is that each array has to be decoded during its manufacture so that the positions of each probe are known; the first 29 nucleotides of the probe sequence are reserved for decoding the array during manufacture. An Illumina BeadChip can support the analysis of multiple samples having up to 12 copies of each array on a chip.  The HumanHT-12 v4.0 Expression BeadChip has over 47,000 probes which are based on RefSeq 38 and Unigene\footnote{http://www.illumina.com/Documents/products/datasheets/datasheet\_gene\_exp\_analysis.pdf}.  Being randomly assembled means that Illumina arrays donâ€™t suffer so much from spatially localized artifacts \cite{Kuhn04} additionally the presence of multiple copies of each bead can be used as technical replicate information for Variance Stabilizing Transformation (VST) \cite{Lin01022008}.

\subsection{lumi}

The \texttt{lumi} package \cite{Du01072008}, available in Bioconductor\footnote{http://www.bioconductor.org/packages/release/bioc/html/lumi.html}, is specifically designed to process BeadArrray data providing VST and quality control steps specifically tuned for this technology.  \texttt{lumi} assumes that data has been pre-background corrected in Illumina's BeadStudio or GenomeStudio, although if it has not they provide the \texttt{lumiB} function which mimics the background correction found therein.  Should you want to investigate processing the raw image files from the scanner directly, in order to have control over the background correction and image processing steps then the Bioconductor package \texttt{beadarray} needs to be used.  This however, depends on the BeadScan Illumina microarray scaner software being set to output raw TIFF files, which is not a default setting, at the time the arrays are processed.  Consequently we are focusing on analysis downstream of BeadStudio or GenomeStudio.  Two key features of \texttt{lumi} which give it an advantage over other methods, which are largely based on Affymetrix data processing methods are: VST which takes advantage of multiple beads per probe when transforming the data and outperforms log2 based transformations \cite{Lin01022008} and Robust Spline Normalisation (RSN) which is designed for Illumina data and combines the advantageous features of both quantile normalisation (fast, gene rank order preserving) and loess normalization which is continuous\footnote{http://www.bioconductor.org/packages/2.11/bioc/vignettes/lumi/inst/doc/lumi.pdf}.  

\subsection{Limma}

Whilst \texttt{lumi} handles the transformation, normalisation and quality control of Illumina microarray data, in order to obtain a list of differentially expressed genes you need to make use of the \texttt{limma} package.  This has advantages over t-tests since it makes use of linear models and empirical Bayes methods in order to determine significant differentially expressed genes \cite{Smyth04}\cite{Smyth05}.  The starting point for analysis in \texttt{limma} is a matrix of expression values from \texttt{lumi} this is accessed via the \texttt{exprs} method provided by the core \texttt{Biobase} package of Bioconductor. 
